nilearn/connectome/connectivity_matrices.py

```diff
@@ -206,15 +207,26 @@ def sym_to_vec(symmetric):
     symmetric : numpy.ndarray, shape (..., n_features, n_features)
         Input array.
 
+    keep_diagonal : boolean, optional
+        Whether to keep or not the coefficient in the diagonal of the matrix.
+        Default is True.
+
     Returns
     -------
-    output : numpy.ndarray, shape (..., n_features * (n_features + 1) / 2)
-        The output flattened lower triangular part of symmetric.
+    output : numpy.ndarray
+        The output flattened lower triangular part of symmetric. Shape is
+        (..., n_features * (n_features + 1) / 2) if keep_diagonal is True and
+        (..., (n_features - 1) * n_features / 2) if it is False.
+
+
     """
-    scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
-    np.fill_diagonal(scaling, 1.)
-    tril_mask = np.tril(np.ones(symmetric.shape[-2:])).astype(np.bool)
-    return symmetric[..., tril_mask] * scaling[tril_mask]
+    if keep_diagonal:
+        scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
```

Comment:
On 28/06/2016 11:17, Gael Varoquaux wrote:
>
> In nilearn/connectome/connectivity_matrices.py 
> <https://github.com/nilearn/nilearn/pull/1153#discussion_r68723710>:
>
> >      """
> > -    scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
> > -    np.fill_diagonal(scaling, 1.)
> > -    tril_mask = np.tril(np.ones(symmetric.shape[-2:])).astype(np.bool)
> > -    return symmetric[..., tril_mask] * scaling[tril_mask]
> > +    if keep_diagonal:
> > +        scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
> > + scaling = sqrt(2) * np.ones(symmetric.shape[-2:]) I proprose to 
> remove this scaling, which is horribly counter-intuitive. It might be 
> available under an 'isometric' option if there is any reason to keep 
> it, but I don't see any use case.
> I seem to remember that it is fairly important to have the metric to 
> really be the Cramer-Rao metric. That said, we could do it the other 
> way around, and apply 1/sqrt(2) on the diagonal, which would be 
> probably more intuitive.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nilearn/nilearn/pull/1153/files/c79280449fff7c06e0fc5cdb645ad7f5d309ad92#r68723710>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAOT1hZNtyTwjMeWvz1Sxo1DhMuyUE-Nks5qQOaTgaJpZM4I9wJA>.
>
Sounds like a reasonable compromise.